generated/src/aws-cpp-sdk-s3/source/S3ErrorMarshaller.cpp

```diff
@@ -127,27 +127,35 @@ AWSError<Aws::Client::CoreErrors> S3ErrorMarshaller::Marshall(const Aws::Http::H
   XmlDocument doc = XmlDocument::CreateFromXmlStream(body);
   body.seekg(readPointer);
   Aws::String bodyError;
+  Aws::String requestId;
 
   if (doc.WasParseSuccessful() &&
       !doc.GetRootElement().IsNull() && 
       doc.GetRootElement().GetName() == Aws::String("Error")) 
   {        
       //check if the first node fetched has no children
       auto messageNode = doc.GetRootElement().FirstChild("Message") ;
-      if(!messageNode.IsNull() && !messageNode.HasChildren())
```

Comment:
previously we were ensuring leaf node but now we just get from the root node.
If say the text message is empty, do we need to recurse to find the first non empty message?

Reply:
> previously we were ensuring leaf node

no we weren't otherwise this would have worked. the child of a empty node `HasChildren` is true with a empty value. i.e.

```cpp
#include <aws/core/Aws.h>
#include <aws/core/utils/xml/XmlSerializer.h>

using namespace Aws;
using namespace Aws::Utils::Xml;

auto main() -> int
{
    SDKOptions options;
    InitAPI(options);
    {
        const auto doc = XmlDocument::CreateFromXmlString("<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>InvalidAction</Code><Message>We encountered an internal error. Please try again.</Message><RequestId>656c76696e6727732072657175657374</RequestId><HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId></Error>");
        assert(doc.WasParseSuccessful());
        auto node = doc.GetRootElement().FirstChild("Message");
        std::cout << node.GetText() << std::endl;
        if (node.HasChildren())
        {
            const auto thechild = node.FirstChild();
            std::cout << thechild.GetText() << std::endl;
        }
    }
    Aws::ShutdownAPI(options);
}

```